src/evagram/database/input_tool.py

```diff
@@ -0,0 +1,174 @@
+            "experiment_name": experiment_name,
+            "owner_id": owner_id
+        }
+        insert_table_record(cur, experiment_obj, "experiments")
+        return add_current_experiment(cur, experiment_name, owner_id)
+
+
+def add_plot(cur, experiment_path, observation_name, plot_filename, experiment_id):
+    plot_file_path = os.path.join(
+        experiment_path, observation_name, plot_filename)
+
+    with open(plot_file_path, 'rb') as file:
+        dictionary = pickle.load(file)
+
+    # extract the div and script components
+    div = dictionary['div']
+    script = dictionary['script']
+
+    # parse filename for components variable name, channel, and group name
+    filename_no_extension = os.path.splitext(plot_filename)[0]
+    plot_components = filename_no_extension.split("_")
+
+    var_name = plot_components[0]
+    channel = plot_components[1] if plot_components[1] != '' else None
+    group_name = plot_components[2]
+
+    # insert observation, variable, group dynamically if not exist in database
+    cur.execute("SELECT observation_id FROM observations WHERE observation_name=%s",
+                (observation_name,))
+    new_observation = len(cur.fetchall()) == 0
+    cur.execute(
+        """SELECT variable_id FROM variables WHERE variable_name=%s
+        AND (channel=%s OR channel IS NULL)""",
+        (var_name, channel))
+    new_variable = len(cur.fetchall()) == 0
+    cur.execute("SELECT group_id FROM groups WHERE group_name=%s", (group_name,))
+    new_group = len(cur.fetchall()) == 0
+
+    if new_observation:
+        observation_obj = {
+            "observation_name": observation_name,
+        }
+        insert_table_record(cur, observation_obj, "observations")
+
+    if new_variable:
+        variable_obj = {
+            "variable_name": var_name,
+            "channel": channel
+        }
+        insert_table_record(cur, variable_obj, "variables")
+
+    if new_group:
+        group_obj = {
+            "group_name": group_name
+        }
+        insert_table_record(cur, group_obj, "groups")
+
+    # get the observation, variable, group ids
+    cur.execute("SELECT observation_id FROM observations WHERE observation_name=%s",
+                (observation_name,))
+    observation_id = cur.fetchone()[0]
+    cur.execute(
+        """SELECT variable_id FROM variables WHERE variable_name=%s
+        AND (channel=%s OR channel IS NULL)""",
+        (var_name, channel))
+    variable_id = cur.fetchone()[0]
+    cur.execute("SELECT group_id FROM groups WHERE group_name=%s", (group_name,))
+    group_id = cur.fetchone()[0]
+
+    # create plot object
+    plot_obj = {}
+    plot_obj["div"] = div
+    plot_obj["script"] = script
+    plot_obj["experiment_id"] = experiment_id
+    plot_obj["observation_id"] = observation_id
+    plot_obj["group_id"] = group_id
+    plot_obj["variable_id"] = variable_id
+
+    # insert plot to database
+    insert_table_record(cur, plot_obj, "plots")
+
+    return 0
```

Comment:
Great job! Good to see the add plot function! Again, I don't think you need to return 0 here